### PR TITLE
Battle log: split long phrases into two lines

### LINKED ISSUE #6357: Long phrases in the battle log
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

Some phrases do not fit in the width of the combat log. (especially in translations)
long_phrase

It could be reproduced in "Battle only" mode: Attack many master swordsmen with other many master swordsmen.
изображение

### Describe the possible solution

_Originally posted by @zenseii in https://github.com/ihhub/fheroes2/issues/6165#issuecomment-1339686868_ :
>This happens in many other languages and I honestly think we should just make it so that these strings wrap to the next line, or make them cut with a [...] part at the end like we do for campaign awards and bonuses.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by Districh-ru**
Fix #6357 by splitting long phrases into two lines:
изображение

For more tests I temporarily reduced the allowed text width (not in this PR, only on my PC - only for test):
изображение

### CODE REVIEW

**Review by Branikolog — APPROVED** [review]

File: src/fheroes2/battle/battle_interface.cpp [hunk 1]
```diff
@@ -735,28 +735,47 @@ namespace Battle
...
+            // Check if message string fits the log width.
+            fheroes2::Text text( str, fheroes2::FontType::normalWhite() );
+            size_t pos = std::string::npos;
+
+            while ( text.width() > battleLogElementWidth ) {
+                // Find the maximum allowed phrase to fit the line. We assume that all words are separated by a space.
+                pos = str.rfind( ' ', pos - 1 );
+
+                // There should not be such a large word.
+                assert( pos != std::string::npos );
```
Comment (ihhub) [inline src/fheroes2/battle/battle_interface.cpp:747]: I am not sure that this is always true. We could simply break from this loop.

File: src/fheroes2/battle/battle_interface.cpp [hunk 2]
```diff
@@ -735,28 +735,47 @@ namespace Battle
...
+
+                text.set( str.substr( 0, pos ), fheroes2::FontType::normalWhite() );
+            }
+
+            if ( pos == std::string::npos ) {
+                _messages.push_back( std::move( str ) );
```
Comment (ihhub) [inline src/fheroes2/battle/battle_interface.cpp:753]: How about the case when indeed a word can be longer than area width? Should we use `fitToOneRow()` and `text()` methods here to get the proper string?

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Districh-ru , I left 2 questions here. Could you please take a look at them?

File: src/fheroes2/battle/battle_interface.cpp (hunk 1 above)
Comment (Districh-ru) [reply src/fheroes2/battle/battle_interface.cpp]: Hi, @ihhub !
Theoretically there might be a very long word: https://en.wikipedia.org/wiki/Longest_word_in_English
I've added an easy logic for a such case.

In original game there are no such words and when adding new phrases and translating them we should avoid using extra-large words that do not fit even the battle log. :))

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@Districh-ru , thank you so much for this improvement!

**Pull request merged**